src/backpack.js

```diff
@@ -15,6 +15,11 @@ export default class Backpack {
    * @returns {*}
    */
   forceFallback(e){
+
+    const extraData = this.getState();
+    this.setState('default', true);
+    CD.setExtraData(extraData);
```

Comment:
I was under the impression that currently when we call `CD.setExtraData()` we pass an object that appends new keys to the `extra_data` object, and overwrites existing keys.

Is it really necessary to create an `extraData` variable and call `this.setState()` _within_ `forceFallback` if we never see/handle the full `extra_data` object here? Why not just:
```javascript
forceFallback(e) {
  CD.setExtraData({ "default" : true });
}
```

Reply:
I was under the impression that it doesn't update the value and it overwrites the value.

Would need to re-test and if that is the case then yes we can do that without a problem, but ideally we would want to set the state so it follows the same format as everything else just to make it consistent. 

Also want to deter everyone from making multiple `CD.setExtraData` calls within the project and just to do it in the `buildCard` or when we `forceFallback`

Not precious about how we do this as they are just how I would prefer to do it and if we are in agreement that this is a better solution then lets go with that 👍

Reply:
Based on the [cropduster docs](https://github.com/movableink/cropduster#storing-extra-analytics-data) I think we can simplify this

Reply:
Looks like it does not overwrite but it will append to the `extra_data` in the HTML.

However, if we are only going to go down the convention of only calling `CD.setExtraData` once in `buildCard` or `forceFallback` we should get the state first in the `forcefallback` as we may have failed further down the project and have set some information using `setState`. 

For example the project is two CORS request, the first CORS request is successful and we have set some info (such as the product name, title and price) and then in our second request we're getting the stock level, but the stock level is less than 5 so we would collapse the block. 
Rather than just setting `default : true` we should still set the other information that has been passed as it could be useful for reporting. 

I agree it could be simpler maybe just doing `setState` to include the `default : true` and then calling `getState` to just set that (cuts out the creation of the extraData variable).

Reply:
Great suggestion - we definitely want to include as much data as possible about _what_ failed when we build projects that make multiple calls.

Sticking with the idea of calling `CD.setExtraData` only once - why not just instantiate `App` with an `extraData` property? That way `this.settings.extraData` _always_ maintains the current version of the `extraData` variable you created on line 19, and we don't need to call `getState()` before setting.

I think this also solves for @roger-rodriguez 's comment below, even if it's inline with `App`. Thoughts?